pydantic/networks.py

```diff
@@ -279,7 +289,17 @@ class PostgresDsn(AnyUrl):
 
 
 class RedisDsn(AnyUrl):
-    allowed_schemes = {'redis'}
+    allowed_schemes = {'redis', 'rediss'}
+
+    @classmethod
+    def validate_parts(cls, parts: Dict[str, Optional[str]]) -> Dict[str, Optional[str]]:
+        defaults = {
+            'domain': 'localhost' if not (parts['ipv4'] or parts['ipv6']) else None,
+            'port': '6379',
+            'path': '/0',
+        }
+        parts = {k: v if v else defaults.get(k) for k, v in parts.items()}
```

Comment:
probably marginally more efficient and clear to do something like

```py
for key, value in defaults.items():
    if not parts[key]:
        parts[key] = value
```
or similar.